Standardize partition settings

WalterKolczynski-NOAA opened the pull request:
The partition settings in setup are updated so all slurm-based
machines are treated the same. Also fixes an issue where task-
specific partitions were created, but the task definition used
PARTITION_BATCH directly instead.

Refs: #417

WalterKolczynski-NOAA committed:
Standardize partition settings

The partition settings in setup are updated so all slurm-based
machines are treated the same. Also fixes an issue where task-
specific partitions were created, but the task definition used
PARTITION_BATCH directly instead.

Refs: #417

WalterKolczynski-NOAA commented:
@KateFriedman-NOAA (and anyone else that is willing to help): can you help test this on the different machines and setups (cycled/fcst-only). The workflows look right, but I haven't actually tried running.

KateFriedman-NOAA commented:
> @KateFriedman-NOAA (and anyone else that is willing to help): can you help test this on the different machines and setups (cycled/fcst-only). The workflows look right, but I haven't actually tried running.

@WalterKolczynski-NOAA I'll test it everywhere, will take at least a day given queues on some platforms.

One comment after looking at your changes (which look good), the earc section of setup_workflow.py can get the same treatment: https://github.com/NOAA-EMC/global-workflow/blob/93b389e9fbd7cdbef4c822e3435ab2c495efd11c/ush/rocoto/setup_workflow.py#L366:L369

Thanks for this!

WalterKolczynski-NOAA committed:
Standardize partition settings for earc task

Refs: #417

WalterKolczynski-NOAA committed:
Add partition setting for GDAS tasks on slurm

Workflow was failing because the GDAS tasks were trying to use a
non-existent entity setting the task partition.

Refs: #417

KateFriedman-NOAA reviewed (approved):
Changes look good and tests of both cycled and free-forecast mode on WCOSS-Dell (Mars), Hera, and Orion were able to successfully submit jobs.
